sample/sample_test.go

```diff
@@ -0,0 +1,137 @@
+package sample
+
+import (
+	"fmt"
+	"math"
+	"math/rand/v2"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestWeighted(t *testing.T) {
+	idx, err := Weighted(nil).Sample([]float32{float32(math.Inf(-1)), 2, float32(math.Inf(-1)), float32(math.Inf(-1))})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	want := int32(1)
+	if diff := cmp.Diff(want, idx); diff != "" {
```

Comment:
`cmp` should be reserved for nested or structured types - I believe here we're just comparing two numbers?

Reply:
We're using `cmp.Diff` for other structures as well - from below: 
<img width="882" alt="image" src="https://github.com/user-attachments/assets/37263237-bd68-43ac-8a68-5e9715890845" />

Personally I prefer the diff to show the exact failure. Probably overkill for for simple integers but if we're already using, thought it would be fine for it. LMK what you think